### PR TITLE
feat: flexible escrow contract

### PR DESCRIPTION
**Pull request by KolevDarko**
Modified the Escrow contract so that whitelist admins are allowed to adjust emergency and freeze periods.
Also modified the contract to be deployable with Request Deployer.
Then deployed new version on rinkeby with the deployer and using that version as version 0.1.0

### CODE REVIEW

**Review by kevindavee — APPROVED** [review]

File: packages/smart-contracts/src/contracts/ERC20EscrowToPay.sol [hunk 1]
```diff
@@ -26,67 +27,79 @@ contract ERC20EscrowToPay {
...
+     */
     mapping(bytes => Request) public requestMapping;
 
     /**
-    * @notice Modifier checks if msg.sender is the requestpayment payer.
-    * @param _paymentRef Reference of the requestpayment related.
-    * @dev It requires msg.sender to be equal to requestMapping[_paymentRef].payer. 
-    */
+     * @notice Duration of emergency claim period that payee can initiate.
+     The payer can reverse this claim within this period.
```
Comment (alexandre-abrioux) [inline packages/smart-contracts/src/contracts/ERC20EscrowToPay.sol:36]: To align with format from
https://github.com/RequestNetwork/requestNetwork/blob/eb28a76064dd874c812ec97dd5dbb53a845af5b2/packages/smart-contracts/src/contracts/EthConversionProxy.sol#L8-L14

```suggestion
     * @notice Duration of emergency claim period that payee can initiate.
     *         The payer can reverse this claim within this period.
```

File: packages/smart-contracts/src/contracts/ERC20EscrowToPay.sol [hunk 2]
```diff
@@ -26,67 +27,79 @@ contract ERC20EscrowToPay {
...
+     */
+    uint256 public emergencyClaimPeriod = 24 weeks;
+
+    /**
+     * @notice Duration of Escrow freeze period that payer can initiate.
+     This lock is irreversable, once the funds are frozen, payer must wait for the whole period.
```
Comment (alexandre-abrioux) [inline packages/smart-contracts/src/contracts/ERC20EscrowToPay.sol:42]:
```suggestion
     * @notice Duration of Escrow freeze period that payer can initiate.
     *         This lock is irreversable, once the funds are frozen, payer must wait for the whole period.
```

File: packages/smart-contracts/test/contracts/ERC20EscrowToPay.test.ts [hunk 3]
```diff
@@ -213,4 +251,26 @@ describe('Contract: ERC20EscrowToPay', () => {
...
+      const contractEmergencyPeriod = await erc20EscrowToPay.connect(payee).emergencyClaimPeriod();
+      expect(contractEmergencyPeriod).to.be.equal(twoDaysInSeconds);
+    });
+    it('Should be able to adjust freeze period, only admin', async () => {
+      const twoDaysInSeconds = 3600 * 24 * 2;
+      await erc20EscrowToPay.connect(admin).setFrozenPeriod(twoDaysInSeconds);
+      const contractFreezePeriod = await erc20EscrowToPay.connect(payee).frozenPeriod();
+      expect(contractFreezePeriod).to.be.equal(twoDaysInSeconds);
+    });
+    it('Contract creator should not be able to change periods', async () => {
```
Comment (alexandre-abrioux) [inline packages/smart-contracts/test/contracts/ERC20EscrowToPay.test.ts:267]: Maybe also add the same test for a random user, like `payer` or `payee`

**Review by alexandre-abrioux — APPROVED** [review]

**Review by yomarion — COMMENTED** [review]
Can you use `Ownable` instead? Nice PR, very clean 👍

File: packages/smart-contracts/src/contracts/ERC20EscrowToPay.sol [hunk 4]
```diff
@@ -1,15 +1,16 @@
 /// SPDX-License-Identifier: MIT
 pragma solidity ^0.8.4;
 
-import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import "./lib/SafeERC20.sol";
-import "./interfaces/ERC20FeeProxy.sol";
+import './legacy_openzeppelin/contracts/access/roles/WhitelistAdminRole.sol';
```
[the remark this answers is not in the document]
Comment (KolevDarko) [reply packages/smart-contracts/src/contracts/ERC20EscrowToPay.sol:4]: Yes, good idea.
At first I thought that we can't transfer the ownership but now saw that we can.

**Review by yomarion — APPROVED** [review]
Well done

**Pull request merged**